include/stronk/unit_v2.h

```diff
@@ -22,43 +24,67 @@ concept unit_value_like = ::twig::stronk_like<T> && requires { typename T::unit_
 // Implementations
 
 // Tag should either be the unit itself, or a dimensions_like type defining the dimensions of the unit.
-template<typename Tag, template<typename StronkT> typename... SkillTs>
+template<typename Tag, typename ScaleT, template<typename StronkT> typename... SkillTs>
 struct unit
 {
     using dimensions_t = std::conditional_t<dimensions_like<Tag>, Tag, twig::create_dimensions_t<dimension<Tag, 1>>>;
+    using scale_t = ScaleT;
 
     unit() = delete;  // Do not construct this type
 
+    // we recreate ratio to make sure we have the gcd version of the scale.
+    template<typename OtherScaleT>
+    using scaled_t = unit<Tag, std::ratio<OtherScaleT::num, OtherScaleT::den>, SkillTs...>;
+
```

Comment:
> The static data members num and den representing the numerator and denominator are calculated by dividing Num and Denom by their greatest common divisor. However, two std::ratio with different Num or Denom are distinct types even if they represent the same rational number (after reduction). A std::ratio type can be reduced to the lowest terms via its type member: std::ratio<3, 6>::type is std::ratio<1, 2>.

From https://en.cppreference.com/w/cpp/numeric/ratio/ratio

So I think you can do `typename OtherScaleT::type`.

Could we `require std::same_as<ScaleT, typename ScaleT::type>` on the unit template just to avoid weird behaviour down the line?